oxidation/crates/parsec_core_fs/src/storage/chunk_storage.rs

```diff
@@ -0,0 +1,404 @@
+        chunks::table
+            .select(count_star())
+            .first(conn)
+            .map_err(|_| FSError::QueryTable("chunks: get_nb_blocks"))
+    }
+
+    fn get_total_size(&self) -> FSResult<i64> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        chunks::table
+            .select(coalesce_total_size())
+            .first(conn)
+            .map_err(|_| FSError::QueryTable("chunks: get_total_size"))
+    }
+
+    fn is_chunk(&self, chunk_id: ChunkID) -> FSResult<bool> {
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        chunks::table
+            .select(count_star())
+            .filter(chunks::chunk_id.eq((*chunk_id).as_ref()))
+            .first::<i64>(conn)
+            .map_err(|_| FSError::QueryTable("chunks: is_chunk"))
+            .map(|res| res > 0)
+    }
+
+    fn get_local_chunk_ids(&self, chunk_ids: &[ChunkID]) -> FSResult<Vec<ChunkID>> {
+        let bytes_id_list = chunk_ids
+            .iter()
+            .map(|chunk_id| (**chunk_id).as_ref())
+            .collect::<Vec<_>>();
+
+        let mut res = Vec::with_capacity(chunk_ids.len());
+
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        for bytes_id_list_chunk in bytes_id_list.chunks(SQLITE_MAX_VARIABLE_NUMBER) {
+            res.append(
+                &mut chunks::table
+                    .select(chunks::chunk_id)
+                    .filter(chunks::chunk_id.eq_any(bytes_id_list_chunk))
+                    .load::<Vec<u8>>(conn)
+                    .map_err(|_| FSError::QueryTable("chunks: get_local_chunk_ids"))?
+                    .into_iter()
+                    .map(|chunk_id| {
+                        <[u8; 16]>::try_from(&chunk_id[..])
+                            .map_err(|_| FSError::QueryTable("chunks: corrupted chunk_id"))
+                    })
+                    .collect::<Result<Vec<_>, _>>()?
+                    .into_iter()
+                    .map(ChunkID::from)
+                    .collect(),
+            )
+        }
+
+        Ok(res)
+    }
+
+    fn get_chunk(&self, chunk_id: ChunkID) -> FSResult<Vec<u8>> {
+        let accessed_on = DateTime::now().get_f64_with_us_precision() as f32;
+
+        let conn = &mut *self.conn().lock().expect("Mutex is poisoned");
+        let changes =
+            diesel::update(chunks::table.filter(chunks::chunk_id.eq((*chunk_id).as_ref())))
+                .set(chunks::accessed_on.eq(accessed_on))
+                .execute(conn)
+                .map_err(|_| FSError::UpdateTable("chunks: get_chunk"))?
+                > 0;
+
+        if !changes {
+            return Err(FSError::LocalMiss(*chunk_id));
+        }
+
+        let ciphered = chunks::table
+            .select(chunks::data)
+            .filter(chunks::chunk_id.eq((*chunk_id).as_ref()))
+            .first::<Vec<u8>>(conn)
+            .map_err(|_| FSError::QueryTable("chunks: get_chunk"))?;
```

Comment:
could be useful to save the error returned by `first::<vec<u8>>(conn)` for debug purpose